contractsV5/KyberNetwork.sol

```diff
@@ -546,17 +544,139 @@ contract KyberNetwork is Withdrawable, Utils, IKyberNetwork, ReentrancyGuard {
+                amountSoFarWithNetworkFee += splitAmount;
+                tradeData.destAmountWithNetworkFee += calcDstQty(splitAmount, ETH_DECIMALS, tradeData.ethToToken.decimals, rate);
+                
+                //calculate split amount with both network and custom platform fee
+                splitAmount = (i == tradeData.ethToToken.splitValuesBps.length - 1) ? 
+                (tradeData.tradeWei - tradeData.networkFeeWei - tradeData.platformFeeWei - amountSoFarWithNetworkAndCustomFee) 
+                : tradeData.ethToToken.splitValuesBps[i] * (tradeData.tradeWei - tradeData.networkFeeWei - tradeData.platformFeeWei) / BPS;
+                amountSoFarWithNetworkAndCustomFee += splitAmount;
+                tradeData.actualDestAmount = calcDstQty(splitAmount, ETH_DECIMALS, tradeData.ethToToken.decimals, rate);
+            }
+        } else {
+            // else, search best reserve and its corresponding dest amount
+            // Have to search with tradeWei minus fees, because that is the actual src amount for ETH -> token trade
+            // Return the rate instead, because destAmountNoFee will be lesser than expected, since we don't use full tradeWei
+            (tradeData.ethToToken.addresses[0], tradeData.ethToToken.rates[0], isPayingFees) = searchBestRate(
+                tradeData.ethToToken.addresses,
+                ETH_TOKEN_ADDRESS, 
+                dest, 
+                tradeData.tradeWei - tradeData.networkFeeWei - tradeData.platformFeeWei,
+                tradeData.takerFeeBps
+            );
+            //store chosen reserve into tradeData
+            tradeData.ethToToken.splitValuesBps[0] = BPS;
+            tradeData.destAmountNoFee = calcDstQty(tradeData.tradeWei, ETH_DECIMALS, tradeData.ethToToken.decimals, tradeData.ethToToken.rates[0]);
+
+            // add to feePayingReservesBps if reserve is fee paying
+            if (isPayingFees) {
+                tradeData.networkFeeWei += tradeData.tradeWei * tradeData.takerFeeBps / BPS;
+                tradeData.feePayingReservesBps += BPS; //max percentage amount for ETH -> token
+                tradeData.numFeePayingReserves ++;
+            }
+
+            //calculate destAmountWithNetworkFee and actualDestAmount
+            tradeData.destAmountWithNetworkFee = calcDstQty(tradeData.tradeWei - tradeData.networkFeeWei, ETH_DECIMALS, tradeData.ethToToken.decimals, tradeData.ethToToken.rates[0]);
+            tradeData.actualDestAmount = calcDstQty(tradeData.tradeWei - tradeData.networkFeeWei - tradeData.platformFeeWei, ETH_DECIMALS, tradeData.ethToToken.decimals, tradeData.ethToToken.rates[0]);
+        }
+        // calc final rate
+        tradeData.rateWithNetworkFee = calcRateFromQty(srcAmount, tradeData.destAmountWithNetworkFee, tradeData.tokenToEth.decimals, tradeData.ethToToken.decimals);
```

Comment:
function is too long
solhin will surely complain.

can we have one function to serve the main loop in a generic way?

anyway.
don't have to fix for now.

Reply:
ok, TODO: refactor a bit hahaha

Reply:
Done!